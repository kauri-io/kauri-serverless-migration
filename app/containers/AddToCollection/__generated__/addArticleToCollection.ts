/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceIdentifierInput } from "./../../../../__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: addArticleToCollection
// ====================================================

export interface addArticleToCollection_addCollectionResource {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface addArticleToCollection {
  addCollectionResource: addArticleToCollection_addCollectionResource | null;
}

export interface addArticleToCollectionVariables {
  id?: string | null;
  sectionId?: string | null;
  resourceId?: ResourceIdentifierInput | null;
  position?: number | null;
}
