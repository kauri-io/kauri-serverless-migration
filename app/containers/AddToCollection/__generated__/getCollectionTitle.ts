/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getCollectionTitle
// ====================================================

export interface getCollectionTitle_getCollection {
  __typename: "CollectionDTO";
  name: string | null;
}

export interface getCollectionTitle {
  getCollection: getCollectionTitle_getCollection | null;
}

export interface getCollectionTitleVariables {
  id?: string | null;
}
