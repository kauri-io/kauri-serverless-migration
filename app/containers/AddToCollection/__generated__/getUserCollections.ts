/* tslint:disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getUserCollections
// ====================================================

export interface getUserCollections_searchCollections_content_sections_resources_CommunityDTO {
  __typename:
    | "CommunityDTO"
    | "PublicUserDTO"
    | "CommentDTO"
    | "CommunityMemberDTO"
    | "TemplateDTO"
    | "CuratedListDTO"
    | "CollectionDTO"
    | "SearchResultDTO"
    | "UserDTO";
}

export interface getUserCollections_searchCollections_content_sections_resources_ArticleDTO {
  __typename: "ArticleDTO";
  id: string | null;
  version: number | null;
}

export type getUserCollections_searchCollections_content_sections_resources =
  | getUserCollections_searchCollections_content_sections_resources_CommunityDTO
  | getUserCollections_searchCollections_content_sections_resources_ArticleDTO;

export interface getUserCollections_searchCollections_content_sections {
  __typename: "SectionDTO";
  id: string | null;
  name: string | null;
  resources:
    | (getUserCollections_searchCollections_content_sections_resources | null)[]
    | null;
}

export interface getUserCollections_searchCollections_content {
  __typename: "CollectionDTO";
  id: string | null;
  name: string | null;
  sections:
    | (getUserCollections_searchCollections_content_sections | null)[]
    | null;
}

export interface getUserCollections_searchCollections {
  __typename: "ResponsePage_CollectionDTO";
  content: (getUserCollections_searchCollections_content | null)[] | null;
}

export interface getUserCollections {
  searchCollections: getUserCollections_searchCollections | null;
}

export interface getUserCollectionsVariables {
  userId?: string | null;
}
