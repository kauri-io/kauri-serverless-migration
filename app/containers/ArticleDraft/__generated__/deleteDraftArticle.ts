/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: deleteDraftArticle
// ====================================================

export interface deleteDraftArticle_cancelArticle {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface deleteDraftArticle {
  cancelArticle: deleteDraftArticle_cancelArticle | null;
}

export interface deleteDraftArticleVariables {
  id?: string | null;
  version?: number | null;
}
