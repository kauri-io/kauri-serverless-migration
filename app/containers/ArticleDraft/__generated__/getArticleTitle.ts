/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getArticleTitle
// ====================================================

export interface getArticleTitle_getArticle {
  __typename: "ArticleDTO";
  title: string | null;
}

export interface getArticleTitle {
  getArticle: getArticleTitle_getArticle | null;
}

export interface getArticleTitleVariables {
  id?: string | null;
  version?: number | null;
}
