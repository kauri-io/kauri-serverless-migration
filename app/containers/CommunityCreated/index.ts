import CollectionCreated from './View'
import { routeChangeAction } from '../../lib/Epics/RouteChangeEpic'
import { connect } from 'react-redux'
import { graphql } from 'react-apollo'
import { flowRight as compose } from 'lodash'
import withLoading from '../../lib/with-loading'
import { getCommunity as getCommunityQuery } from '../../queries/Community'
import {
    getCommunityVariables,
    getCommunity_getCommunity,
} from '../../queries/__generated__/getCommunity'

const mapStateToProps = () => {
    return {}
}

export default compose(
    connect(
        mapStateToProps,
        { routeChangeAction }
    ),
    graphql<getCommunity_getCommunity, getCommunityVariables>(
        getCommunityQuery,
        {
            options: ({ id }) => ({ variables: { id } }),
        }
    ),
    withLoading()
)(CollectionCreated)
