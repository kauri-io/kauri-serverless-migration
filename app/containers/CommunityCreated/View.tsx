import React from 'react'
import styled from 'styled-components'
import CommunityCard from '../../components/Card/CommunityCard'
import Link from 'next/link'
import Head from 'next/head'
import PrimaryButton from '../../components/Button/PrimaryButton'
import { Title2, BodyCard } from '../../components/Typography'
import { getCommunity } from '../../queries/__generated__/getCommunity'

interface IProps {
    data: getCommunity
    routeChangeAction: (route: string) => void
    type: 'created' | 'updated'
}

const Container = styled.section`
    display: flex;
    height: calc(100vh - 76px);
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${props => props.theme.colors.textPrimary};
    > :first-child {
        margin-bottom: ${props => props.theme.space[1]}px;
        color: white;
    }
    > :nth-child(2) {
        margin-bottom: ${props => props.theme.space[3]}px;
        color: white;
    }
    > :nth-child(3) {
        margin-bottom: ${props => props.theme.space[3]}px;
    }
`

const Row = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
`

const CommunityCreated: React.FunctionComponent<IProps> = props => {
    const { routeChangeAction, type } = props
    const copy = type === 'updated' ? 'updated' : 'live'
    const subtitleCopy =
        copy === 'live' ? (
            <Row>
                <BodyCard>
                    Your community will be live soon, we are just creating it!
                </BodyCard>
                <BodyCard>This usually takes less than 10 seconds.</BodyCard>
            </Row>
        ) : (
            <BodyCard>Your community is now updated</BodyCard>
        )

    if (props.data.getCommunity) {
        const {
            approved,
            id,
            description,
            name,
            avatar,
            attributes,
            // tags,
        } = props.data.getCommunity

        const articleCount =
            approved &&
            approved.reduce((current, next) => {
                if (next && next.__typename === 'ArticleDTO') {
                    current += 1
                }
                return current
            }, 0)

        const collectionCount =
            approved &&
            approved.reduce((current, next) => {
                if (next && next.__typename === 'CollectionDTO') {
                    current += 1
                }
                return current
            }, 0)

        return (
            <Container>
                <Head>
                    <title>{`Community ${copy} - Kauri`}</title>
                </Head>
                <Title2>Community</Title2>
                {subtitleCopy}
                <CommunityCard
                    cardHeight={310}
                    description={String(description)}
                    name={String(name)}
                    imageURL={attributes && attributes.background}
                    logo={avatar}
                    articleCount={String(articleCount)}
                    collectionCount={String(collectionCount)}
                    linkComponent={(childrenProps: React.ReactElement<any>) => (
                        <Link href={`/community/${String(id)}`}>
                            {childrenProps}
                        </Link>
                    )}
                />
                <PrimaryButton
                    onClick={() =>
                        routeChangeAction(`/community/${String(id)}`)
                    }
                >
                    View Community
                </PrimaryButton>
            </Container>
        )
    }
    return null
}

export default CommunityCreated
