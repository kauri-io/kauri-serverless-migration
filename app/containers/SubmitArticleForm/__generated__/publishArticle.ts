/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceIdentifierInput } from "./../../../../__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: publishArticle
// ====================================================

export interface publishArticle_publishArticle {
  __typename: "MutationResponse";
  hash: string | null;
}

export interface publishArticle {
  publishArticle: publishArticle_publishArticle | null;
}

export interface publishArticleVariables {
  id?: string | null;
  version?: number | null;
  owner?: ResourceIdentifierInput | null;
  signature?: string | null;
  updateComment?: string | null;
}
