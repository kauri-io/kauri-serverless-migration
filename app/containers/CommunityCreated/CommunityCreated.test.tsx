import { shallow } from 'enzyme'
import CommunityCreated, { IProps } from './View'
import { Community } from '../../mocks'
import { getCommunity_getCommunity } from '../../queries/__generated__/getCommunity'

describe('containers/CommunityCreated/View', () => {
    it('should match the snapshot', () => {
        const props = {
            data: {
                getCommunity: Community.data
                    .getCommunity as getCommunity_getCommunity,
            },
            routeChangeAction: jest.fn(),
            type: 'updated' as IProps['type'],
        }
        const wrapper = shallow(<CommunityCreated {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
