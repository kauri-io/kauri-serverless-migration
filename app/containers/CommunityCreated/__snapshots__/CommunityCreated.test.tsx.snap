// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/CommunityCreated/View should match the snapshot 1`] = `
<View__Container>
  <Head>
    <title>
      Community updated - Kauri
    </title>
  </Head>
  <Typography__styledComponent>
    Community
  </Typography__styledComponent>
  <Typography__component>
    Your community is now updated
  </Typography__component>
  <CommunityCard
    articleCount="6"
    cardHeight={310}
    collectionCount="0"
    description="Java is one of the worlds most popular programming languages used in global-scale applications by developers everywhere. This community brings together developers and projects at the forefront of bringing Ethereum support to this rich language ecosystem."
    id="5d2f30daaba2920001c82409"
    imageURL="https://api.kauri.io:443/ipfs/QmPyoXQaK9uA1oedsptssr1EhYRBF1A9vrnypbKAkMhuxQ"
    linkComponent={[Function]}
    logo="https://api.kauri.io:443/ipfs/QmQ9to7bfbR6ADTG2tAKVxWTNw8o3YbwJ1LzdpEJeanLBX"
    name="Java Ethereum"
  />
  <PrimaryButtonComponent
    onClick={[Function]}
  >
    View Community
  </PrimaryButtonComponent>
</View__Container>
`;
